File: 019)JavaInstagramCloneFirebase/app/src/main/java/com/example/javainstagramclonefirebase/adapter/PostAdapter.java
package com.example.javainstagramclonefirebase.adapter;

import android.content.Context;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import androidx.annotation.NonNull;
import androidx.databinding.DataBindingUtil;
import androidx.recyclerview.widget.RecyclerView;

import com.example.javainstagramclonefirebase.R;
import com.example.javainstagramclonefirebase.databinding.RecyclerRowBinding;
import com.example.javainstagramclonefirebase.model.Post;
import com.squareup.picasso.Picasso;

import java.util.List;

public class PostAdapter extends RecyclerView.Adapter<PostAdapter.PostHolder>{
    private Context mContext;
    private List<Post> postList;

    public PostAdapter(Context mContext, List<Post> postList) {
        this.mContext = mContext;
        this.postList = postList;
    }

    @NonNull
    @Override
    public PostHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(mContext);
        RecyclerRowBinding binding = DataBindingUtil.inflate(layoutInflater, R.layout.recycler_row, parent, false);
        return new PostHolder(binding);
    }

    @Override
    public void onBindViewHolder(@NonNull PostHolder holder, int position) {
        Post post =postList.get(position);
        RecyclerRowBinding binding = holder.binding;
        binding.textViewRVUserEmail.setText(post.getEmail());
        binding.textViewRVComment.setText(post.getComment());
        Picasso.get().load(post.getDownloadUrl()).into(binding.imageViewRV);
    }

    @Override
    public int getItemCount() {
        if (postList == null){
            return 0;
        }else {
            return postList.size();
        }
    }

    public class PostHolder extends RecyclerView.ViewHolder{
        private RecyclerRowBinding binding;

        public PostHolder(RecyclerRowBinding binding) {
            super(binding.getRoot());
            this.binding = binding;
        }
    }
}

File: 059)KotlinInstagram/app/src/main/java/com/elyesasimsek/kotlininstagram/adapter/PostAdapter.kt
package com.elyesasimsek.kotlininstagram.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.databinding.DataBindingUtil
import androidx.recyclerview.widget.RecyclerView
import com.elyesasimsek.kotlininstagram.R
import com.elyesasimsek.kotlininstagram.databinding.RecyclerRowBinding
import com.elyesasimsek.kotlininstagram.model.Post
import com.squareup.picasso.Picasso

class PostAdapter (
    private var context: Context,
    private var postList: List<Post>
): RecyclerView.Adapter<PostAdapter.postHolder>() {
    class postHolder(val binding: RecyclerRowBinding): RecyclerView.ViewHolder(binding.root)

    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): postHolder {
        val layoutInflater = LayoutInflater.from(context)
        val binding = DataBindingUtil.inflate<RecyclerRowBinding>(layoutInflater, R.layout.recycler_row, parent, false)
        return postHolder(binding)
    }

    override fun getItemCount(): Int {
        if (postList == null){
            return 0;
        }else{
            return postList.size
        }
    }

    override fun onBindViewHolder(holder: postHolder, position: Int) {
        val post = postList[position]
        val binding = holder.binding

        binding.textViewRVUserEmail.text = post.email
        binding.textViewRVComment.text = post.comment
        Picasso.get().load(post.downloadUrl).into(binding.imageViewRV)

    }
}